### PR TITLE
Fixed some bugs with the new artifact filter processing

### PR DESCRIPTION
**Pull request by jnettesheim**
## One line description of pull request
Fixing some bugs when processing from log2timeline.py

## Description:
There were some errors while processing from log2timeline.py, fixing these by refactoring passing of filter arguments.

**Related issue (if applicable):** fixes #<plaso issue number here>
Related to #1313

## Notes:
All contributions to Plaso undergo [code
review](https://github.com/log2timeline/plaso/wiki/Codereview). This makes sure
that the code has appropriate test coverage and conforms to the Plaso [style
guide](https://github.com/log2timeline/plaso/wiki/Style-guide).

One of the maintainers will examine your code, and may request changes.

## Checklist:
  - [ ] Local tests pass
  - [ ] Tests on TravisCI pass
  - [ ] Codacy passes (or flags issues that you think are acceptable)
  - [ ] Coveralls indicates test coverage is sufficient

If [new dependencies](https://github.com/log2timeline/plaso/wiki/Adding-a-new-dependency) are required:
 - [ ] l2tdevtools has been updated

### CODE REVIEW

File: plaso/cli/helpers/artifact_definitions.py [hunk 1]
```diff
@@ -127,6 +127,7 @@ def ParseOptions(cls, options, configuration_object):
             'Unable to read artifact definitions from: {0:s} with error: '
             '{1!s}').format(custom_artifacts_path, exception))
 
+    setattr(configuration_object, '_artifacts_registry', registry)
```
Comment (joachimmetz) [inline plaso/cli/helpers/artifact_definitions.py]: This is not a configuration value I opt we solve this differently.

File: tests/cli/helpers/artifact_filters.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ class ArtifactFiltersArgumentsHelperTest(cli_test_lib.CLIToolTestCase):
 
   _EXPECTED_OUTPUT = """\
 usage: cli_helper.py [--artifact_filters ARTIFACT_FILTERS]
-                     [--artifact_filters_file ARTIFACT_FILTERS]
+                     [--artifact_filters_file ARTIFACT_FILTERS_FILE]
```
Comment (joachimmetz) [inline tests/cli/helpers/artifact_filters.py]: I opt: ARTIFACT_FILTERS_FILE => PATH

File: plaso/cli/helpers/artifact_definitions.py (hunk 1 above)
Comment (jnettesheim) [reply plaso/cli/helpers/artifact_definitions.py]: Understood, just to be completely clear though, it was this way prior to my earlier artifact filters PR.  I removed it per your suggestion, but noticed today that it broke some pre-processing, I thought best to revert it back to original state, but can take a closer look now.  Specifically, 

      extraction_engine.PreprocessSources(
          self._artifacts_registry, self._source_path_specs,
          resolver_context=self._resolver_context)

in plaso/cli/extraction_tool.py

Comment (joachimmetz) [reply plaso/cli/helpers/artifact_definitions.py]: Let's add a comment to change this then for now.

Comment (jnettesheim) [reply plaso/cli/helpers/artifact_definitions.py]: Went ahead and moved this over to using the BuildArtifactsRegistry method, no longer passing registry in the configuration value.  Should be good now.

File: tests/cli/helpers/artifact_filters.py (hunk 2 above)
Comment (jnettesheim) [reply tests/cli/helpers/artifact_filters.py]: Done.

**Comment by jnettesheim** [comment]
Updated, PTAL @joachimmetz

**Pull request merged**